packages/ui-select/src/Select/index.tsx

```diff
@@ -458,53 +442,50 @@ class Select extends Component<SelectProps> {
     return groupChildren
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'data' implicitly has an 'any' type.
-  renderList(data) {
+  renderList(
+    data: Pick<
+      SelectableRender,
+      'getListProps' | 'getOptionProps' | 'getDisabledOptionProps'
+    >
+  ) {
     const { getListProps, getOptionProps, getDisabledOptionProps } = data
-    const {
-      isShowingOptions,
-      optionsMaxWidth,
-      visibleOptionsCount,
-      children
-    } = this.props
+    const { isShowingOptions, optionsMaxWidth, visibleOptionsCount, children } =
+      this.props
 
     let lastWasGroup = false
-    const viewProps = isShowingOptions
+
+    const viewProps: Partial<ViewProps> = isShowingOptions
       ? {
           display: 'block',
           overflowY: 'auto',
-          // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
-          maxHeight: this._optionHeight * visibleOptionsCount,
+          maxHeight: this._optionHeight * visibleOptionsCount!,
           maxWidth: optionsMaxWidth || this.width,
           background: 'primary',
-          // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'node' implicitly has an 'any' type.
-          elementRef: (node) => (this._listView = node)
+          elementRef: (node: Element | null) => (this._listView = node)
         }
       : { maxHeight: 0 }
 
     return (
-      //@ts-expect-error TODO: fix this
       <View {...viewProps}>
         <Options
           {...getListProps({ as: 'ul', elementRef: this.handleListRef })}
         >
           {isShowingOptions
-            ? // @ts-expect-error ts-migrate(7030) FIXME: Not all code paths return a value.
-              Children.map(children, (child, index) => {
+            ? Children.map(children, (child, index) => {
                 if (!child || !matchComponentTypes(child, [Group, Option])) {
                   return // ignore invalid children
                 }
                 if (matchComponentTypes(child, [Option])) {
```

Comment:
We should modernize this code in the future, there is lots of boilerplate going on just for these comparisons (if I understand correctly, this code is the sole reason for Select.Group and Select.Option's existence)

Reply:
It is indeed pretty weird, but we already started the talks about writing a new Select component and retire this one...